src/supervised/ordinal.jl

```diff
@@ -0,0 +1,49 @@
+"""
+    OrdinalMarginLoss <: SupervisedLoss
+
+Modifies a margin loss `loss` to be used on an ordinal domain with
+number of levels `nlevels`. It treats each level as an integer between
+1 and `nlevels`, inclusive, and penalizes output according to the sum of
+level thresholds crossed relative to target
+
+Assumes target is encoded in an Index encoding scheme where levels are
+numbered between 1 and `nlevels`
+"""
+struct OrdinalMarginLoss <: SupervisedLoss
+    loss::MarginLoss
+    nlevels::Int
+end
+
+for fun in (:value, :deriv, :deriv2)
+    @eval @fastmath function ($fun)(loss::OrdinalMarginLoss, target::Number, output::Number)
+        retval = 0
```

Comment:
I think a problem you may have is that initializing this variable with an `Int` may cause type instability since it is likely that `value` or `deriv` return some float